### DISCUSSION TITLE
Can I set color level overrides across all colors for things like hover states

### DISCUSSION THREAD

**Discussion by mattaningram** [mantinedev/mantine#3998] [Q&A]
I want to change the background color that gets set for `Button` and `ActionIcon` using the `subtle` variant to use level `[2]` instead of `[1]` when hovering for example. But I don't want to have to put in overrides for every possible color. Is there a way to tell Mantine to use a different color level for all of a particular style in a variant?

**Comment by rtivital** [comment]
You can change `theme.colors` object instead

**Reply by mattaningram** [reply]
I should probably explain better. I have already replaced all of the `theme.colors` with my own overrides. I am hoping to tell components like `Button` and `ActionIcon` to use a different shade of each color (for example `[2]` instead of `[1]`) for all `light` variants. I know I can go through each possible color and variant and override it in the `ThemeProvider` but I was hoping there was a simpler way where I could just tell it what color level to use across all colors in a variant.

**Reply by rtivital** [reply]
Nope, currently it is not possible. You can find colors resolving loogic here – https://github.com/mantinedev/mantine/blob/master/src/mantine-styles/src/theme/functions/fns/variant/variant.ts

**Reply by ldalves** [reply]
Is this true for ALL elements, eg borders on Inputs components as well?

**Reply by rtivital** [reply]
You can customize styles of all components with Styles API on MantineProvider.

**Reply by ldalves** [reply]
Yes and it works great. But then you have to do it for each separate component and your overall design might not line up in the end. But I agree, customizing with Styles API on MantineProvider is the best option at the moment. :)

**Comment by gdostie** [comment]
I am looking into doing the "same" (changing `subtle` variant hover color shade from 0 to 1) thing. From [the documentation](https://mantine.dev/theming/theme-object/), I see the theme can receive functions
![image](https://user-images.githubusercontent.com/35579930/232086686-1dd2ff54-efce-4718-ae8b-f5b9c3d48c22.png)
Since `variant` is one of those function, I am wondering if this could be leveraged to override what we want.
